### ISSUE TITLE
Is there a way to ignore some folders from the files list

### ISSUE THREAD

**Issue by Lawrence-Dube** [doxygen/doxygen#7426]
For example, we have to include specific platform for each kind of documentation (licensing stuff), so we include them all individually like so :
![image](https://user-images.githubusercontent.com/2318469/70081332-5024e000-15d6-11ea-9e32-801bb0482a4d.png)

But with 1.8.15 it started to include them in the File list at the bottom of the TOC
![image](https://user-images.githubusercontent.com/2318469/70081402-79457080-15d6-11ea-9ae8-ab2530df94ac.png)

I would like to be able to filter Folder from the File list (red circle)

Anyone has any workaround?

**Comment by albert-github** [comment]
Only thing I can, quickly. think of is to modify the list in the appropriate `.js` files(s) in the HTML output directory. I'm not sure whether this has negative  side effects.
One thing one has to think about is that in the "Files" / "File List" etc. in the "top blue bar" there might still be references to the directories / files

**Comment by Lawrence-Dube** [comment]
Could you go into more details ? Looking at the `.js` files in the HTML output directory and I can't find anything related to that.

Ideally, a feature to filter the file list would be a lot better than having to do some post-process to remove the files.

**Comment by albert-github** [comment]
For my test I had the information in the `files_dup.js`. So look in the `.js` file for terms like `private`, `Doxygen` (but try also `doxygen) etc. Without an example and the used doxygen version it is hard to tell, so
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.
- Please also specify the doxygen version used.

**Comment by Lawrence-Dube** [comment]
`But with 1.8.15` is the version. I'll look if I can find that file

**Comment by Lawrence-Dube** [comment]
`dynsections.js` and `jquery.js` are the only 2 js files i can find

**Comment by albert-github** [comment]
I was talking about the `GENERATE_TREEVIEW` part in the HTML output.
Are you talking about CHM output ?
In the later case the files is `index.hhc` in the html directory that you have to edit and rerun the chm generation process again like: `"C:\Program Files (x86)\HTML Help Workshop\hhc.exe" index.hhp`

**Comment by Lawrence-Dube** [comment]
`GENERATE_TREEVIEW` is set to false
Yes, I'm talking about the CHM output.

Regardless of the output though, having to post-process doxygen's output seems like something we would end up doing anyway.

**Comment by albert-github** [comment]
What kind of further post-processing are you thinking about?
It always good to know what kind of wishes there are, but due to the generality (and complexity) of the package it is not always possible to fulfill all. One thing that is always necessary, especially to know what has been done, is to have a, small, self contained example.

**Comment by Lawrence-Dube** [comment]
No, I'm saying that what you are proposing is to have some post-processing to remove the files from list hhc or js files depending on output kind.

**Comment by Lawrence-Dube** [comment]
The best thing would be to have a FILE_LIST_FILTER or  FILE_LIST_EXCLUDE where we can just filter/exclude folders from showing there

**Comment by MuellerSeb** [comment]
I also got this problem, since I have a `doc` folder with markdown files for documentation, it is showing as an empty folder in the file list.

**Comment by albert-github** [comment]
@MuellerSeb which version of doxygen did you use?

**Comment by aspieln3r** [comment]
@albert-github How I encountered this issue is by doing the following:
Create a separate main.dox file for the 'main page' of html output and store it in a folder without any files related to source code.
add this folder to INPUTS list and set RECURSIVE to YES 
If files list, an entry pointing to this folder will appear and I'd like to hide this empty folder coming in file list if possible

**Comment by aspieln3r** [comment]
https://llvm.org/doxygen/files.html
This issue in wild. You can see a empty docs entry on top of files list. LLVM stores their main page in a folder named docs

**Comment by albert-github** [comment]
@aspieln3r 
- "The issue in the wild" looks like to use doxygen 1.8.17, so what happens with the current doxygen version (1.9.5)?
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by julienalofs** [comment]
This issue only occurs when the EXTRACT_ALL parameter is set to YES. 

Here follows a small sample that reproduce the issue:
[doxy.tar.gz](https://github.com/doxygen/doxygen/files/9850211/doxy.tar.gz)

You can see in the generated files.html the folders where the .dox file is located. 
Setting the EXTRACT_ALL parameter to NO fixes the issue.

**Comment by albert-github** [comment]
The fact that `EXTRACT_ALL` looks like to have an influence is due to fact that with `EXTRACT_ALL=YES` everything is shown documented and non documented identities.

**Comment by julienalofs** [comment]
> The fact that `EXTRACT_ALL` looks like to have an influence is due to fact that with `EXTRACT_ALL=YES` everything is shown documented and non documented identities.

Why do we see only the folders and not the files?

**Comment by albert-github** [comment]
Looks like doxygen does not (yet?) remove the directories as well.
